applications/convection_diffusion_application/custom_elements/adjoint_diffusion_element.h

```
@@ -0,0 +1,262 @@
+///@{
+
+/// Basic element for the ajdoint diffusion problem.
+/** The element supports arbitrary variables in the primal problem by defining CONVECTION_DIFFUSION_SETTINGS in the ProcessInfo.
+ *  Note that the adjoint variable is hard-coded to ADJOINT_HEAT_TRANSFER for now.
+ */
+template< class PrimalElement >
+class AdjointDiffusionElement: public PrimalElement
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Counted pointer of AdjointDiffusionElement
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(AdjointDiffusionElement);
+
+    using IndexType = typename PrimalElement::IndexType;
+    using GeometryType = typename PrimalElement::GeometryType;
+    using NodesArrayType = typename PrimalElement::NodesArrayType;
+    using MatrixType = typename PrimalElement::MatrixType;
+    using VectorType = typename PrimalElement::VectorType;
+    using EquationIdVectorType = typename PrimalElement::EquationIdVectorType;
+    using DofsVectorType = typename PrimalElement::DofsVectorType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    AdjointDiffusionElement(IndexType NewId, typename GeometryType::Pointer pGeometry);
+
+    AdjointDiffusionElement(IndexType NewId, typename GeometryType::Pointer pGeometry,  Properties::Pointer pProperties);
+
+    /// Destructor.
+    ~AdjointDiffusionElement() override;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    Element::Pointer Create(IndexType NewId, NodesArrayType const& ThisNodes,  Properties::Pointer pProperties) const override;
+
+    Element::Pointer Create(IndexType NewId, typename GeometryType::Pointer pGeom,  Properties::Pointer pProperties) const override;
+
+    void CalculateLocalSystem(MatrixType& rLeftHandSideMatrix, VectorType& rRightHandSideVector, ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateRightHandSide(VectorType& rRightHandSideVector, ProcessInfo& rCurrentProcessInfo) override;
+
+    void GetValuesVector(Vector& rValues, int Step = 0) override;
```

Default value in virtual function can be misleading 
```suggestion
    void GetValuesVector(Vector& rValues, int Step) override;
```